File: antlr/src/main/java/com/rescribe/antlr/parse/schema/Variable.java
package com.rescribe.antlr.parse.schema;

import java.util.ArrayList;
import java.util.List;
import lombok.*;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class Variable {
  String name;
  String type;
  Location location;
  List<Comment> comments = new ArrayList<>();
}

File: api/src/schema/variable.ts
import { ObjectType, Field } from 'type-graphql';
import Comment from './comment';
import Location from './location';

@ObjectType({ description: 'variable' })
export default class Variable {
  @Field({ description: 'name' })
  name: string;
  @Field({ description: 'type' })
  type: string;
  @Field(_type => Location, { description: 'location' })
  location: Location;
  @Field(_type => [Comment], { description: 'comments' })
  comments: Comment[];
}
